### ISSUE TITLE
Error message printed in the middle of the output for another file

### ISSUE THREAD

**Issue by iafisher** [sharkdp/bat#946]
**What version of `bat` are you using?**
0.9.0

**Describe the bug you encountered:**
When invoked with two file paths, one that exists and one that doesn't, bat will sometimes print a file not found error message in the middle of the contents of the other file:

![image](https://user-images.githubusercontent.com/7120244/80266114-26388c80-864f-11ea-947e-97a8a815c3dc.png)

The bug seems to be rather sensitive to the name and contents of the file. For example, if I name the file `a.txt` instead of `package.json`, or if I replace the JSON contents with a bunch of periods, there's no bug.

Here's the text of the file:

```
{
  "name": "...",
  "version": "1.0.0",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "license": "MIT",
  "private": true,
  "dependencies": {
  }
}
```

**Describe what you expected to happen?**
I expected the error message for the missing file to be either before or after the contents of the other file.

**How did you install `bat`?**
I'm afraid I don't specifically recall, but probably through apt-get.

---

system
------

**$ uname -srm**
Linux 4.15.0-96-generic x86_64  

**$ lsb_release -a**
No LSB modules are available.  
Distributor ID:	Ubuntu  
Description:	Ubuntu 18.04.4 LTS  
Release:	18.04  
Codename:	bionic  

bat
---

**$ bat --version**
bat 0.9.0  

**$ env**
PAGER=less  
error: Found argument '--cache-dir' which wasn't expected, or isn't valid in this context

USAGE:
    bat [OPTIONS] [FILE]...
    bat <SUBCOMMAND>

For more information try --help

bat_config
----------

bat_wrapper
-----------

No wrapper script.

bat_wrapper_function
--------------------

No wrapper function for 'bat'.

No wrapper function for 'cat'.

tool
----

**$ less --version**
less 487 (GNU regular expressions)

**Comment by sharkdp** [comment]
Thank you for your detailed report.

That version of `bat` is quite old, but that behavior is still present in newer versions. The problem is the following:

- The normal output of `bat` is piped to a pager (`less`).
- The error messages are printed to STDERR.

If the non-existent file comes second, the pager is already running when the error message is printed to STDERR in the background. This can lead to the effect you've seen.

The behavior should not be present without the pager (`--paging=never`).

I think there could be an easy way to fix this by redirecting the error messages to STDOUT for cases where we print the pager anyway. This way, error messages would still be present if someone does this:
```
bat package.json s | some-other-program
bat package.json s > some-file
```

**Comment by sharkdp** [comment]
Fixed in [v0.15.0](https://github.com/sharkdp/bat/releases/tag/v0.15.0).

**Comment by iafisher** [comment]
Thank you for fixing it so quickly!
